prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/metrics/impl/MetricsManager.java

```diff
@@ -0,0 +1,332 @@
+     */
+    private void reporterCallback(final BulkGenericDto dtos) {
+        for (Consumer<BulkGenericDto> consumer : consumers.values()) {
+            try {
+                consumer.accept(dtos);
+            } catch (Throwable e) {
+                logger.error("Error updating SensiNact metrics: {}", e.getMessage(), e);
+            }
+        }
+    }
+
+    /**
+     * Checks if metrics are enabled for that name
+     *
+     * @param name Metrics name
+     * @return True if the metric is allows
+     */
+    private boolean isEnabled(final String name) {
+        return isActive && registry != null && (activeMetrics.isEmpty() || activeMetrics.contains(name));
+    }
+
+    @Override
+    public void enableMetrics() {
+        isActive = true;
+
+        // Restart the console reporter if allowed
+        if (allowConsoleReporter) {
+            if (consoleReporter != null) {
+                consoleReporter.close();
+                consoleReporter = null;
+            }
+
+            consoleReporter = ConsoleReporter.forRegistry(registry).convertDurationsTo(TimeUnit.MILLISECONDS)
+                    .convertRatesTo(TimeUnit.SECONDS).build();
+            consoleReporter.start(updateRate, TimeUnit.SECONDS);
+        } else if (consoleReporter != null) {
+            consoleReporter.close();
+            consoleReporter = null;
+        }
+
+        // Restart the callback reporter
+        if (callbackReporter != null) {
+            callbackReporter.close();
+            callbackReporter = null;
+        }
+
+        if (callbackReporter == null) {
+            callbackReporter = new CallbackReporter(this::reporterCallback, registry, metricsProvider);
+            callbackReporter.start(updateRate, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void disableMetrics() {
+        isActive = false;
+
+        if (consoleReporter != null) {
+            consoleReporter.close();
+            consoleReporter = null;
+        }
+
+        if (callbackReporter != null) {
+            callbackReporter.close();
+            callbackReporter = null;
+        }
+
+        // Clear metrics
+        clear();
+    }
+
+    @Override
+    public void clear() {
+        // Remove all metrics but the gauges
+        final Set<String> toRemove = new HashSet<>(registry.getNames());
+        toRemove.removeAll(registry.getGauges().keySet());
+        registry.removeMatching((name, metric) -> toRemove.contains(name));
+    }
+
+    @Override
+    public void enableMetrics(String... names) {
+        if (names != null) {
+            activeMetrics.addAll(Arrays.asList(names));
+        }
+    }
+
+    @Override
+    public void disableMetrics(String... names) {
+        if (names != null) {
+            activeMetrics.removeAll(Arrays.asList(names));
+        }
+    }
+
+    @Override
+    public IMetricTimer withTimer(String name) {
+        if (!isEnabled(name)) {
+            return new DummyTimer(name);
```

Comment:
Note that timer is different from the counter/histogram comments below. This is being obtained for every call so a dummy object works well.